### DISCUSSION TITLE
Supporting Unit of Measurement (UoM) as built in

### DISCUSSION THREAD

**Discussion by welly87** [questdb/questdb#1151] [Ideas]
Hi All, 

One thing that missing i think from all of the timeseries database is UoM. In my previous works on [O&G](https://www.petrolink.com/) and [Energistics](https://www.energistics.org/) we have that as core concept in the domain. 

It's have it's own [standard](https://www.energistics.org/energistics-unit-of-measure-standard/).

I think all of the sensors or IoT etc have some kind of measurement? As we need to align that first to do computation..

Thought?

**Comment by newskooler** [comment]
Can you please expand a bit more on that? How would that work? I cannot picture it myself at the moment.

**Reply by bluestreak01** [reply]
I am _assuming_ here that is something similar to `symbol` type that we already have, but as a standalone type.

**Reply by welly87** [reply]
@bluestreak01 so there's a similar features? where i can read the details about this type? thanks

**Reply by bsmth** [reply]
@welly87 there's a short page about this type here: https://questdb.io/docs/concept/symbol/

The purpose of the type is to store tag-like strings (`USD` / `celcius`) which repeat often through a data set internally as integers. This reduces the storage space required for the type and allows faster filtering and search as you don't need to perform constant string comparison.
